daemon/algod/api/server/v2/handlers.go

```diff
@@ -1139,6 +1141,98 @@ func (v2 *Handlers) GetApplicationByID(ctx echo.Context, applicationID uint64) e
 	return ctx.JSON(http.StatusOK, response)
 }
 
+// GetApplicationBoxes returns the box names of an application
+// (GET /v2/applications/{application-id}/boxes)
+func (v2 *Handlers) GetApplicationBoxes(ctx echo.Context, applicationID uint64, params generated.GetApplicationBoxesParams) error {
+	appIdx := basics.AppIndex(applicationID)
+	ledger := v2.Node.LedgerForAPI()
+	lastRound := ledger.Latest()
+	keyPrefix := logic.MakeBoxKey(appIdx, "")
+
+	// cast param.Max to zero, if param.Max points to some value, we keep the value
+	castedMax := nilToZero(params.Max)
+	maxBoxThreshold := v2.Node.Config().MaxAPIBoxPerApplication
+
+	// compute `maxReturnNum` from `castedMax` and `maxBoxThreshold := v2.Node.Config().MaxAPIBoxPerApplication`
+	// `maxReturnNum == 0` for returning everything
+	// - `castedMax == 0` and `maxBoxThreshold == 0`,
+	//       return everything
```

Comment:
Your proposal sounds consistent with the way unlimited assets were implemented.

So if users have a huge number of boxes and need to fetch them all with algod, they would have to make sure the max box threshold configuration is large enough. Depending on how much data there is that might mean they need to provision a machine with extra ram.

Reply:
@tzaffi My opinion is to leave as is.

* The current implementation provides safeguards and I don't see a convincing rationale for disallowing customization by an end user.  
* In my opinion, the root issue at-hand is lack of pagination.  Since, we've previously agreed algod pagination is out-of-scope to the PR, I think the implementation is a reasonable middle ground.

Reply:
Right, this "gap" is an explicitly opted into "gap".  You (algod runner) would have to run your `algod` with the configuration parameter set to 0.  It's your machine! Presumably you didn't give the API token out to knuckleheads.